examples/audio/melgan_spectrogram_inversion.py

```diff
@@ -0,0 +1,574 @@
+            gen_audio_wave = generator(x_batch_train, training=True)
+
+            # Generating the features using the discriminator
+            fake_pred = discriminator(y_batch_train)
+            real_pred = discriminator(gen_audio_wave)
+
+            # Calculating the generator losses
+            gen_loss = generator_loss(real_pred, fake_pred)
+            fm_loss = feature_matching_loss(real_pred, fake_pred)
+
+            # Calculating final generator loss
+            gen_fm_loss = gen_loss + 10 * fm_loss
+
+            # Calculating the discriminator losses
+            disc_loss = discriminator_loss(real_pred, fake_pred)
+
+        # Calculating and applying the gradients for generator and discriminator
+        grads_gen = gen_tape.gradient(gen_fm_loss, generator.trainable_weights)
+        grads_disc = disc_tape.gradient(disc_loss, discriminator.trainable_weights)
+        gen_optimizer.apply_gradients(zip(grads_gen, generator.trainable_weights))
+        disc_optimizer.apply_gradients(zip(grads_disc, discriminator.trainable_weights))
+
+        self.gen_loss_tracker.update_state(gen_fm_loss)
+        self.disc_loss_tracker.update_state(disc_loss)
+
+        return {"gen_loss": self.gen_loss_tracker.result(), "disc_loss": self.disc_loss_tracker.result()}
+
+"""
+## Training
+
+The paper suggests that the training with dynamic shapes takes around 400,000 steps (~500
+epochs). For
+this example, we will run it only for a single epoch (819 steps). Longer training time
+(greater than 300 epochs) will almost certainly provide better results.
+"""
+
+gen_optimizer = keras.optimizers.Adam(
+    LEARNING_RATE_GEN, beta_1=0.5, beta_2=0.9, clipnorm=1
+)
+disc_optimizer = keras.optimizers.Adam(
+    LEARNING_RATE_DISC, beta_1=0.5, beta_2=0.9, clipnorm=1
+)
+
+# Start training
+generator = create_generator((None, 1))
+discriminator = create_discriminator((None, 1))
+
+mel_gan = MelGAN(generator, discriminator)
+mel_gan.compile(gen_optimizer, disc_optimizer, generator_loss, feature_matching_loss, discriminator_loss)
+mel_gan.fit(
+    train_dataset.shuffle(200).batch(BATCH_SIZE).prefetch(tf.data.AUTOTUNE), epochs=1
+)
+
+"""
+## Testing the model
+"""
+
+"""
+The trained model can now be used for real time text-to-speech translation tasks.
+To test how fast the MelGAN inference can be, let us take a sample audio mel-spectrogram
+and convert it. Note that the actual model pipeline will not include the `MelSpec` layer
+and hence this layer will be disabled during inference. The inference input will be a
+mel-spectrogram processed similar to the `MelSpec` layer configuration.
+
+For testing this, we will create a randomly uniformly distributed tensor to simulate the
+behavior of the inference pipeline.
+"""
+
+# Sampling a random tensor with the shape [50, 80] to mimic a spectrogram
+audio_sample = tf.random.uniform([1, 50, 80])
+
+"""
+Timing the inference speed of a single sample.
+"""
+%%timeit
+generator(audio_sample, training=False)
```

Comment:
You should use `predict()`. Using `__call__` outside of a tf.function will be slow since it's executing eagerly.

Reply:
Switched to `predict` instead of `__call__`